### ISSUE TITLE
Location header not set when navigateTo is used for redirection

### ISSUE THREAD

**Issue by mathieumagalhaes** [nuxt/nuxt#18592]
### Environment

- Operating System: `Darwin`
- Node Version:     `v19.0.0`
- Nuxt Version:     `3.1.1`
- Nitro Version:    `2.1.0`
- Package Manager:  `yarn@1.22.11`
- Builder:          `vite`
- User Config:      `app`, `runtimeConfig`, `imports`, `components`, `modules`, `i18n`, `css`, `postcss`, `routeRules`, `nitro`, `vite`
- Runtime Modules:  `@nuxtjs/i18n@8.0.0-beta.8`, `@pinia/nuxt@0.4.6`
- Build Modules:    `-`

### Reproduction

Add the following to the nuxt.config file:
```
  modules: [
    '@nuxtjs/i18n',
  ],
  i18n: {
    strategy: 'prefix',
    customRoutes: 'config',
    detectBrowserLanguage: {
      useCookie: true,
      cookieKey: 'LANGUAGE',
      alwaysRedirect: true,
    },
    locales: [
      {
        code: 'nl',
        name: 'Nederlands',
      },
      {
        code: 'fr',
        name: 'Français',
      },
      {
        code: 'en',
        name: 'English',
      },
    ],
    defaultLocale: 'en',
    vueI18n: {
      legacy: false,
      locale: 'en',
      fallbackLocale: 'nl',
      messages: {
        nl: { ... },
        fr: { ... },
        en: { ... },
      },
    },
  },
```

Surf to the root route `/`

### Describe the bug

**tldr;** Using navigateTo for redriection causes redirection (Status 302) without Location header.
As is the case when using @nuxtjs/i18n module with `alwaysRedirect: true`

Going to the root route `/` we get a 302 response without Location header.
The presence of the Location header is no absolute requirement.
The absence of this header would have questionable practicability though.

I've came across this with healthy check software that did not cope with this.

I think a solution would be to add the header before this line: https://github.com/nuxt/nuxt/blob/de4086f6edabefc1a46ee08a099c3065f24f6fe5/packages/nuxt/src/app/composables/router.ts#L104

**Comment by manniL** [comment]
Sounds reasonable! PR welcome 😋

**Comment by thijsw** [comment]
Related to #18561 ?

**Comment by manniL** [comment]
Absolutely! Consolidating here 👍
